# Analysis of the Gun Violence Dataset

# Part 1. &ndash; Problem Understanding & Data Understanding

Authors: Kacper Neumann, Stanisław Apanasiewicz, Jan Matusiak

----

## 1. The Dataset

### 1.1 General Information

The [Gun Violence Data](https://www.kaggle.com/datasets/jameslko/gun-violence-data) dataset contains detailed information about gun violence incidents in the United States, available in CSV format. It is based on records collected by the [Gun Violence Archive (GVA)](https://www.gunviolencearchive.org/), an independent organization created to provide a comprehensive database incidents in the U.S.

The CSV file contains data for all recorded gun violence incidents in the U.S. between January 2013 and March 2018.

In [16]:
import kagglehub
import pandas as pd
import os

path = kagglehub.dataset_download('jameslko/gun-violence-data')
df = pd.read_csv(os.path.join(path, 'gun-violence-data_01-2013_03-2018.csv'))

In [17]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 239677 entries, 0 to 239676
Data columns (total 29 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   incident_id                  239677 non-null  int64  
 1   date                         239677 non-null  str    
 2   state                        239677 non-null  str    
 3   city_or_county               239677 non-null  str    
 4   address                      223180 non-null  str    
 5   n_killed                     239677 non-null  int64  
 6   n_injured                    239677 non-null  int64  
 7   incident_url                 239677 non-null  str    
 8   source_url                   239209 non-null  str    
 9   incident_url_fields_missing  239677 non-null  bool   
 10  congressional_district       227733 non-null  float64
 11  gun_stolen                   140179 non-null  str    
 12  gun_type                     140226 non-null  str    
 13  incident_c

In [18]:
df['incident_id'].nunique()

239677

There are 239677 samples described with 29 attributes each. There are no duplicate rows.

### 1.2 Attributes

TODO

## 2. Data Mining Goals & Success Criteria

TODO